datanode/datanode.go

```diff
@@ -268,6 +268,18 @@ func (d *dataNode) startDebugServer() {
 
 	d.opts.InstrumentOptions().Logger().Infof("Starting HTTP server on dbg-port %d", d.opts.ServerConfig().DebugPort)
 	d.opts.InstrumentOptions().Logger().Fatal(http.ListenAndServe(fmt.Sprintf(":%d", d.opts.ServerConfig().DebugPort), debugRouter))
+
+	for {
```

Comment:
actually this will never get executed

also this for loop will block on the select